src/bit-systems/link-hover-menu.ts

```diff
@@ -0,0 +1,51 @@
+import { defineQuery, enterQuery, entityExists } from "bitecs";
+import type { HubsWorld } from "../app";
+import { Interacted, Link, LinkHoverMenu, LinkHoverMenuItem, HoveredRemoteRight } from "../bit-components";
+import { anyEntityWith } from "../utils/bit-utils";
+
+const NULL_EID = 0;
+
+const hoveredQuery = defineQuery([HoveredRemoteRight, Link]);
+const hoveredEnterQuery = enterQuery(hoveredQuery);
+const hoveredMenuItemQuery = defineQuery([HoveredRemoteRight, LinkHoverMenuItem]);
+const clickedMenuItemQuery = defineQuery([Interacted, LinkHoverMenuItem]);
+
+export function linkHoverMenuSystem(world: HubsWorld) {
+  // Assumes always only single LinkHoverMenu entity exists.
+  const menuEid = anyEntityWith(world, LinkHoverMenu)!;
+  const menuObject = world.eid2obj.get(menuEid)!;
+
+  // Save Link object eid in LinkHoverMenu when hovered.
+  hoveredEnterQuery(world).forEach((eid: number) => {
+    LinkHoverMenu.targetObject[menuEid] = eid;
+    const targetObject = world.eid2obj.get(eid)!;
+    targetObject.add(menuObject);
+  });
+
+  // Check if the cursor it hovered on Link object or Link menu button object.
+  const hovered = hoveredQuery(world).length > 0 || hoveredMenuItemQuery(world).length > 0;
```

Comment:
We have now implemented something similar to this in 3 different spots in 3 slightly different ways (video menu, pdf menu, and this link menu). This is probably ok for burndown but we should really think about the general pattern here. In this and video menu we tag the sub items with a component, in PDFMenu we check if the hovered thing is itself or is a child of an interesting object. I think I like the second one better design wise, but walking the hierarchy every frame sounds slightly concerning perf wise.

Reply:
> We have now implemented something similar to this in 3 different spots in 3 slightly different ways

Yes, I noticed it. I focused on making it just work in this cycle. In next or later cycle, I hope we can work more on interactive stuffs and update the code to handle more consistently.

Reply:
What I have been thinking is placing invisible plane or box and showing/hiding the menu depending on whether it is raycasted. It can avoid the hierarchy traverse. And there may be gaps between buttons or button and object. Even if curson points to the gaps the menu keeps appearing. Perhaps it is what users expect.